src/navigationGuards.ts

```diff
@@ -294,6 +301,10 @@ export function extractComponentsGuards(
             const resolvedComponent = isESModule(resolved)
               ? resolved.default
               : resolved
+            // __asyncLoader is added by defineAsyncCompoent
+            if (__DEV__ && '__asyncLoader' in resolvedComponent) {
```

Comment:
Is this necessary? What does it change?

Reply:
Is this accounting for `() => defineAsyncComponent(...)`? If it is, it should warn instead and it should also have a test

Reply:
It is also  accounting for `() => import(() => defineAsyncComponet())`

Reply:
It is also accounting for 
```js
// MyPage.vue
export default defineAsyncComponent(...)
//router.js
() => import('./MyPage.vue')
```